### ISSUE TITLE
Run postExport after EAS update

### ISSUE THREAD

**Issue by bentburton** [expo/eas-cli#1522]
### Summary

I am trying to use sentry-expo and you are supposed to add a postPublish hook as instructed here: https://docs.expo.dev/guides/using-sentry/. This is so you can upload your source maps to sentry after releasing new code.

This doesn't seem to do anything when running eas update.

### Managed or bare?

Bare

### Environment

  expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 12.6
      Shell: 5.8.1 - /bin/zsh
    Binaries:
      Node: 14.18.1 - ~/.nvm/versions/node/v14.18.1/bin/node
      Yarn: 1.22.19 - ~/Code/glasspassapp/node_modules/.bin/yarn
      npm: 8.9.0 - ~/.nvm/versions/node/v14.18.1/bin/npm
      Watchman: 2022.10.31.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.11.3 - /opt/homebrew/lib/ruby/gems/3.0.0/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 22.1, iOS 16.1, macOS 13.0, tvOS 16.1, watchOS 9.1
    IDEs:
      Android Studio: 2021.2 AI-212.5712.43.2112.8609683
      Xcode: 14.1/14B47b - /usr/bin/xcodebuild
    npmPackages:
      expo: ^46.0.0 => 46.0.17 
      react: 18.0.0 => 18.0.0 
      react-dom: 18.0.0 => 18.0.0 
      react-native: 0.69.6 => 0.69.6 
    npmGlobalPackages:
      eas-cli: 2.6.0
      expo-cli: 6.0.8
    Expo Workflow: bare

expo doctor
Expected package @expo/config-plugins@^5.0.0
Found invalid:
  @expo/config-plugins@4.1.5
  (for more info, run: npm why @expo/config-plugins)

### Reproducible demo or steps to reproduce from a blank project

Create new project
Install sentry-expo
Add the post publish hook
See that it doesn't run

**Comment by flexbox** [comment]
@bentburton Is your issue still running?

I have the same feeling using Expo with EAS in a managed workflow

```
"expo": "~47.0.3",
"@sentry/react-native": "4.9.0",
"sentry-expo": "~6.0.0",
```

**Comment by MrChip53** [comment]
This issue should be retitled to something like `Run postExport after EAS update`. This is not a bug but a feature request as it's currently not implemented.

Relevant file and approximate line of where to add the hook: https://github.com/expo/eas-cli/blob/ea553a10253f050036a4b49f24c303a24cdc6ad3/packages/eas-cli/src/commands/update/index.ts#L441

If I have time I can look into making a PR in the future to add functionality.

**Comment by flexbox** [comment]
@MrChip53 Thank you for the precision it was not crystal clear from my understanding 😄

**Comment by erikkodnar** [comment]
Any updates on this feature?

**Comment by bentburton** [comment]
Any updates here?

**Comment by quinlanj** [comment]
hey @bentburton sorry for the delay here, we've released SDK 50 and have improved our integration to Sentry. A couple things to note:
- Integration should be as simple as `eas update && npx sentry-expo-upload-sourcemaps dist` now
- EAS Update doesnt support `postPublish`, but it should not be required anymore. 

https://expo.dev/changelog/2024/01-18-sdk-50#first-class-expo-support-in-sentry

**Comment by brentvatne** [comment]
to add on to @quinlanj's point here - the reason i introduced `postPublish` many years ago was because with classic updates, `expo-cli publish` did not export a `dist` directory and i decided that instead of adding that, i would pass the manifest, bundle, and sourcemaps through a hook. this was actually not a great idea after all, and when we built EAS Update we made it output the manifest, bundle, and sourcemaps to the `dist` directory instead to make it more explicit and easier to inspect. so, now there is no need for a hook that passes this data along.
